src/frontend/src/components/authenticateBox.ts

```diff
@@ -219,11 +219,17 @@ export const authenticateBoxFlow = async <T, I>({
     }
 
     result satisfies { tag: "recover" };
-    const result2 = await recover();
-    return {
-      newAnchor: false,
-      ...result2,
-    };
+
+    const recoverResult = await recover();
+    if (recoverResult.tag === "ok") {
+      // If an anchor was recovered, then it's _not_ a new anchor
+      return {
+        newAnchor: false,
+        ...recoverResult,
+      };
+    } else {
+      return recoverResult;
```

Comment:
Does the `recoverResult` have a `newAnchor` field?

The type is `LoginFlowResult<T, E = object>`, isn't it? It's no assurance that `newAnchor` is present.

Reply:
only in the success case! that's the idea